ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/HiveRelDistribution.java

```diff
@@ -81,8 +84,16 @@ public RelDistribution apply(TargetMapping mapping) {
       return this;
     }
     List<Integer> newKeys = new ArrayList<>(keys.size());
+
+    // Instead of using a HashMap for lookup newKeys.add(mapping.getTargetOpt(key)); should be called but not all the
+    // mapping supports that. See HIVE-23963. Replace this when this is fixed in calcite.
```

Comment:
If it is meant to be fixed in Calcite then we should create a JIRA and add an entry in `org.apache.hadoop.hive.ql.optimizer.calcite.Bug`. We could even skip the JIRA creation and move this comment in `Bug` as `CALCITE-X0000_fixed` or something similar.

Reply:
@kasakrisz , please create the Calcite JIRA so it is easier to track. Thanks

Reply:
Created: https://issues.apache.org/jira/browse/CALCITE-4166